app/src/display/widgets/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
 
 target_sources_ifdef(CONFIG_ZMK_WIDGET_BATTERY_STATUS app PRIVATE battery_status.c)
 target_sources_ifdef(CONFIG_ZMK_WIDGET_OUTPUT_STATUS app PRIVATE output_status.c)
+
+target_sources(app PRIVATE layer_status.c)
```

Comment:
This needs to be behind a new Kconfig configuration, and only built if that Kconfig is enabled.

For that configuration... It needs to have a `depends on !ZMK_SPLIT_PERIPHERAL` since we don't *have* a layer state on the peripheral side of keyboards.